added prerelease and release section for 1/25 (#1079)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/releasenotes.md

*   Replaced the `CreateCookieWithCookie` method with the `CopyCookie` method.  
*   Added visual hosting support using the [ICoreWebView2CompositionController][ReferenceWin32Icorewebview2controllerViewWebview210773CompositionController] interface, which is created using the new `CreateCoreWebView2CompositionController` method from `ICoreWebView2Environment3`. 

    
##### Bug fixes  

*   Turned off the Microsoft Edge Shopping feature in WebView2.  
*   Turned off the context menu in the PDF viewer when `AreDefaultContextMenusEnabled` is `false`.  \([\#605][GithubMicrosoftedgeWebviewfeedbackIssue605]\).  
*   Fixed a bug that returned `E_NOINTERFACE` when querying `ICoreWebView2` for `ICoreWebView2Experimental`.  \([\#691][GithubMicrosoftedgeWebviewfeedbackIssue691]\).  
*   Fixed a bug that allowed navigation with malformed URIs when `CoreWebView2NavigationStartingEventArgs.Cancel` was set to `false`.  \([\#400][GithubMicrosoftedgeWebviewfeedbackIssue400]\).  
*   Fixed a bug that blocked `window.print()` on pop-up windows with event handlers attached to `NewWindowRequested` events.  \([\#409][GithubMicrosoftedgeWebviewfeedbackIssue409]\).  

> *   Fixed Dynamic DPI issue when moving apps between different monitors. \([\#58][GithubMicrosoftedgeWebviewfeedbackIssue549]\)  

Reviewer: A definition for `GithubMicrosoftedgeWebviewfeedbackIssue58` is required too.
```suggestion
*   Fixed Dynamic DPI issue when moving apps between different monitors. \([\#58][GithubMicrosoftedgeWebviewfeedbackIssue58]\)  
```

Reviewer: A definition for `GithubMicrosoftedgeWebviewfeedbackIssue58` is missing and `[ReferenceWin32IdlCreatecorewebview2environmentwithoptions09538` under 0.9.538 is missing a closing `]`

*   Fixed a bug that `E_NOINTERFACE` is returned when querying `ICoreWebView2` for `ICoreWebView2Experimental`.  \([\#691][GithubMicrosoftedgeWebviewfeedbackIssue691]\).  
*   Fixed a bug that the navigation with malformed uri is not canceled when `CoreWebView2NavigationStartingEventArgs.Cancel` is `false`.  \([\#400][GithubMicrosoftedgeWebviewfeedbackIssue400]\).  
*   Fixed a bug that `window.print()` does not work with a pop-up window when an event handler is attached to `NewWindowRequested` event.  \([\#409][GithubMicrosoftedgeWebviewfeedbackIssue409]\).  
*   Fixed Dynamic DPI issue when moving between monitors. \([\#58][GithubMicrosoftedgeWebviewfeedbackIssue549]\)  
    
##### Promotions  

*   Promoted the following experimental changes to stable.  
    *   Visual Hosting APIs.  
    *   [SetVirtualHostNameToFolderMapping][ReferenceWin32Icorewebview210773PreReleaseSetVirtualHostNameToFolderMapping]  
    *   [TrySuspend and Resume][ReferenceWin32Icorewebview210773PreReleaseTrySuspendResume] 
    *   [DefaultBackgroundColor][ReferenceWin32Icorewebview2controllerViewWebview210773PreReleaseDefaultBackgroundColor]  
    > [!NOTE]

>     > TrySuspend and Resume and DefaultBackgroundColor are APIs that are fast-tracked into stable. There are no experimental versions of these APIs. 

Author: Maybe unclear, these APIs are fast-tracked, which means they did not go through the whole 6weeks in experimental process. Developers may need to know this, even though it is unlikely

> [!NOTE]
> Re-compile your app after updating the NuGet package.  

> [!NOTE]

> > Please note Minimum Edge version is for supporting running WebView. Some APIs are defined in newer interfaces. Please ensure you are using the proper runtime version for prerelease package APIs. You may also use Microsoft Edge Canary channel to ensure the APIs are supported.

Reviewer: ```suggestion
> Please note Minimum Edge version is for supporting running WebView. Some APIs are defined in newer interfaces. Please ensure you are using the proper runtime version for experimental APIs in prerelease packages. You may also use Microsoft Edge Canary channel to ensure the APIs are supported.
```

> [!NOTE]
> Re-compile your app after updating the NuGet package.  

> ## 1.0.774-prerelease  

Reviewer: Prerelease candidate is likely 773.
```suggestion
## 1.0.773-prerelease  
```

Reviewer: Other version instances have to be updated too.

Author: Changed to 773!

Release Date: January 25, 2021  

[NuGet package][NuGetGallery1.0.774-prerelease] \| minimum Microsoft Edge version 86.0.616.0.  

#### General  

> [!IMPORTANT]
>  **Breaking Change**: WebView2 pre-release package 0.9.430 has been deprecated, and will be removed the upcoming release. Please discontinue development with this packages.  

###### Features  
*   Disabled Edge Shopping feature in WebView2. 
*   Added [TrySuspend & Resume][ReferenceWin32Icorewebview210774PreReleaseTrySuspendResume] method that lets you suspend and resume a WebView. 
*   Added [SetVirtualHostNameToFolderMapping][ReferenceWin32Icorewebview210774PreReleaseSetVirtualHostNameToFolderMapping] method that maps between a virtual host name and a folder path. 

> *   Added [DefaultBackgroundColor][ReferenceWin32Icorewebview2controllerViewWebview210774PreReleaseDefaultBackgroundColor] property that lets you set the background color and transparency.   \([\#414][GithubMicrosoftedgeWebviewfeedbackIssue549]\)

Reviewer: Versions and link definitions to be added/updated.
```suggestion
*   Added [DefaultBackgroundColor][ReferenceWin32Icorewebview2controllerViewWebview210774PreReleaseDefaultBackgroundColor] property that lets you set the background color and transparency.   \([\#414][GithubMicrosoftedgeWebviewfeedbackIssue414]\)
```

Reviewer: `GithubMicrosoftedgeWebviewfeedbackIssue414` is undefined.

Author: https://github.com/MicrosoftEdge/WebView2Feedback/issues/414 it works for me?

*   Added [UserAgent][ReferenceWin32Icorewebview2experimentalsettings10774PreReleaseGetUserAgent] property that lets you get and set the User Agent.\([\#122][GithubMicrosoftedgeWebviewfeedbackIssue549]\)

> 

Reviewer: Can you please add one more note?
```suggestion
*   Updated CreateCookieWithCookie to become CopyCookie.
```

*   Fixed a bug that `E_NOINTERFACE` is returned when querying `ICoreWebView2` for `ICoreWebView2Experimental`.  \([\#691][GithubMicrosoftedgeWebviewfeedbackIssue691]\).  
*   Fixed a bug that the navigation with malformed uri is not cancelled when `CoreWebView2NavigationStartingEventArgs.Cancel` is `false`.  \([\#400][GithubMicrosoftedgeWebviewfeedbackIssue400]\).  
*   Fixed a bug that `window.print()` does not work with a pop-up window when an event handler is attached to `NewWindowRequested` event.  \([\#409][GithubMicrosoftedgeWebviewfeedbackIssue409]\).  
*   Fixed Dynamic DPI issue when moving between monitors. \([\#58][GithubMicrosoftedgeWebviewfeedbackIssue549]\)

###### Promotions  
*   Promoted the following experimental changes to stable:
    * Visual Hosting APIs.
    * URI Protocol Scheme.

#### .NET  

###### Bug fixes
* Fixed bug in WPF that was crashing webview on closing window with F4.  \([\#399][GithubMicrosoftedgeWebviewfeedbackIssue399]\)
* Fixed WebView2 initialization screen to be transparent, instead of gray.   \([\#196][GithubMicrosoftedgeWebviewfeedbackIssue196]\)

> ## 1.0.705.51  

Reviewer: Release candidate is likely 705.50.

Author: changed to 705.50!

> [!NOTE]
> Re-compile your app after updating the NuGet package.  

> [!NOTE]

> > The minimum version of Microsoft Edge supports running WebView.  Some APIs are defined in newer interfaces.  Ensure you use the proper runtime version for exerimental APIs in prerelease packages.  To ensure the APIs are supported, you may also use Microsoft Edge Canary channel.  

Reviewer 2: ```suggestion
> The minimum version of Microsoft Edge supports running WebView.  Some APIs are defined in newer interfaces.  Ensure you use the proper runtime version for experimental APIs in prerelease packages.  To ensure the APIs are supported, you may also use Microsoft Edge Canary channel.  
```

## 1.0.773-prerelease  

Release Date: January 25, 2021  

[NuGet package][NuGetGallery1.0.773-prerelease] \| minimum Microsoft Edge version 86.0.616.0.  

#### General  

> [!IMPORTANT]
> **Breaking Change**:  WebView2 pre-release package 0.9.430 is deprecated, and is being removed the upcoming release.  Discontinue development with it.  

###### Features  

*   Added [TrySuspend and Resume][ReferenceWin32Icorewebview210773PreReleaseTrySuspendResume] method to suspend and resume a WebView.  
*   Added [SetVirtualHostNameToFolderMapping][ReferenceWin32Icorewebview210773PreReleaseSetVirtualHostNameToFolderMapping] method that maps between a virtual host name and a directory path.  
*   Added [DefaultBackgroundColor][ReferenceWin32Icorewebview2controllerViewWebview210773PreReleaseDefaultBackgroundColor] property to set the background color and transparency.  \([\#414][GithubMicrosoftedgeWebviewfeedbackIssue414]\)  

> *   Added [UserAgent][ReferenceWin32Icorewebview2experimentalsettings10773PreReleaseGetUserAgent] property to get and set the User Agent.\([\#122][GithubMicrosoftedgeWebviewfeedbackIssue549]\)

Reviewer: ```suggestion
*   Added [UserAgent][ReferenceWin32Icorewebview2experimentalsettings10773PreReleaseGetUserAgent] property to get and set the User Agent.\([\#122][GithubMicrosoftedgeWebviewfeedbackIssue122]\)
```

###### Promotions  
*   Promoted the following experimental changes to stable:
    * Visual Hosting APIs.
    * [SetVirtualHostNameToFolderMapping][ReferenceWin32Icorewebview210774PreReleaseSetVirtualHostNameToFolderMapping]

>     * [TrySuspend & Resume][ReferenceWin32Icorewebview210774PreReleaseTrySuspendResume]

Reviewer: There are no experimental versions of TrySuspend & Resume and DefaultBackgroundColor. They are fast-tracked to public IDL directly.

Author: Yes, but they are stable in the prerelease, so I noted it as a 'promotion', even though it wasn't experimental.  Will add a note.

[DOMContentLoadedAPI]: /microsoft-edge/webview2/reference/win32/icorewebview2_2?view=webview2-1.0.721-prerelease#add_domcontentloaded&preserve-view=true "add_DOMContentLoaded - interface ICoreWebView2_2 | Microsoft Docs"
[WebViewEnvironmentproperty]: /microsoft-edge/webview2/reference/win32/icorewebview2_2?view=webview2-1.0.721-prerelease#get_environment&preserve-view=true "ICoreWebView2CookieManager | Microsoft Docs"

> [GithubMicrosoftedgeWebviewfeedbackIssue196]:  https://github.com/MicrosoftEdge/WebViewFeedback/issues/196 "Feedback repo for MicrosoftEdge/WebViewFeedback Issue 196" 

Reviewer: Missing GithubMicrosoftedgeWebviewfeedbackIssue122

[NuGetGallery1.0.705.50]:  https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.705.50 "NuGet Gallery | Microsoft.Web.WebView2 v1.0.705.50"
[NuGetGallery1.0.773-prerelease]:  https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.773-prerelease "NuGet Gallery | Microsoft.Web.WebView2 v1.0.773 prerelease"  
[ReferenceWin32Icorewebview210773PreReleaseTrySuspendResume]: /microsoft-edge/webview2/reference/win32/icorewebview2_3?view=webview2-1.0.773-prerelease&preserve-view=true#TrySuspend "TrySuspend - interface ICoreWebview2_3 | Microsoft Docs"
[ReferenceWin32Icorewebview2experimentalsettings10773PreReleaseGetUserAgent]: /microsoft-edge/webview2/reference/win32/icorewebview2/icorewebview2experimentalsettingsget_useragent "get_UserAgent - interface ICoreWebView2ExperimentalSettings | Microsoft Docs"
[ReferenceWin32Icorewebview210773PreReleaseSetVirtualHostNameToFolderMapping]: /microsoft-edge/webview2/reference/win32/icorewebview2_3?view=webview2-1.0.773-prerelease&preserve-view=true#SetVirtualHostNameToFolderMapping "SetVirtualHostNameToFolderMapping - interface ICoreWebView2_3 | Microsoft Docs" 

> [ReferenceWin32Icorewebview2controllerViewWebview210773PreReleaseDefaultBackgroundColor]: //microsoft-edge/webview2/reference/win32/icorewebview2controller?view=webview2-1.0.773-prerelease&preserve-view=true#defaultbackgroundcolor "DefaultBackgroundColor - interface ICoreWebView2Controller2 | Microsoft Docs"

Reviewer: ```suggestion
[ReferenceWin32Icorewebview2controllerViewWebview210773PreReleaseDefaultBackgroundColor]: /microsoft-edge/webview2/reference/win32/icorewebview2controller?view=webview2-1.0.773-prerelease&preserve-view=true#defaultbackgroundcolor "DefaultBackgroundColor - interface ICoreWebView2Controller2 | Microsoft Docs"
```

> [ReferenceWin32Icorewebview2controllerViewWebview210773CompositionController]:  //microsoft-edge/webview2/reference/win32/icorewebview2compositioncontroller?view=webview2-1.0.773-prerelease&preserve-view=true

Reviewer: ```suggestion
[ReferenceWin32Icorewebview2controllerViewWebview210773CompositionController]:  /microsoft-edge/webview2/reference/win32/icorewebview2compositioncontroller?view=webview2-1.0.773-prerelease&preserve-view=true
```
